### DISCUSSION TITLE
Let me propagate search terms from related elements to source relations

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/cms#10037] [Ideas]
I'm proposing a new field setting for relation fields called something like "Inherit Search Terms" that would allow source elements to inherit the search terms of target elements. This essentially expands upon the existing functionality of matrix blocks passing their search terms to owner elements.

The keywords generated for relationships are currently just derived from the element titles. 

The goal of this feature is to better allow for nested content to be used in-place of crowded element edit interfaces without sacrificing searchability.

**Comment by AugustMiller** [comment]
I pretty routinely wish there were some more controls for search term generation.

Specifically, I'd love to see some kind of “Search Term Template” that would be interpreted as a Twig object template… or perhaps a larger text-area which then gets "treated" as all keywords do (i.e. stripping HTML, trimming whitespace). Basically a place to output whatever kind of stuff you want—from that field's scalar data, the element's relations, or even combinations of fields (See #7093).

I s'pose what I'm looking for could also be implemented at the Section/Entry Type/Volume/Category Group/etc. level, where "extra" stuff just gets lumped together in a virtual search “attribute.”

**Comment by AugustMiller** [comment]
Following up here, because @lindseydiloreto and I were talking about formatting field output for exports.

The same “search term template” idea could be applied here, i.e. “What should this field’s value look like in exports?”
